packages/synapse-react-client/src/components/ChallengeTeamWizard/ChallengeTeamWizard.tsx

```diff
@@ -1,76 +1,56 @@
-import React, { useCallback, useEffect, useRef, useState } from 'react'
-import StepperDialog, { Step } from '../StepperDialog/StepperDialog'
-import {
-  ErrorResponse,
-  TeamMembershipStatus,
-} from '@sage-bionetworks/synapse-types'
+import React, { useCallback, useMemo, useRef, useState } from 'react'
 import {
   CreateChallengeTeam,
   CreateChallengeTeamHandle,
 } from './CreateChallengeTeam'
 import { SelectChallengeTeam } from './SelectChallengeTeam'
 import { RegistrationSuccessful } from './RegistrationSuccessful'
-import { JoinRequestForm } from './JoinRequestForm'
-import { useSynapseContext } from '../../utils'
 import {
-  addTeamMemberAsAuthenticatedUserOrAdmin,
-  createMembershipRequest,
-} from '../../synapse-client'
+  MembershipRequestForm,
+  MembershipRequestFormHandle,
+} from './MembershipRequestForm'
+import { useSynapseContext } from '../../utils'
 import {
+  useAddMemberToTeam,
   useGetCurrentUserProfile,
   useGetEntityChallenge,
   useGetMembershipStatus,
   useGetUserSubmissionTeams,
 } from '../../synapse-queries'
-import { ANONYMOUS_PRINCIPAL_ID } from '../../utils/SynapseConstants'
-import { Typography } from '@mui/material'
-import { useQueryClient } from '@tanstack/react-query'
+import { Alert, Box, Button, Tooltip, Typography } from '@mui/material'
+import { SignInPrompt, SynapseErrorBoundary } from '../error/ErrorBanner'
+import { noop } from 'lodash-es'
+import { DialogBase } from '../DialogBase'
+import { SynapseSpinner } from '../LoadingScreen/LoadingScreen'
+import {
+  AcceptMembershipInvitationButton,
+  OpenMembershipInvitation,
+} from './OpenMembershipInvitation'
 
-enum StepsEnum {
+enum ChallengeTeamWizardStep {
   SELECT_YOUR_CHALLENGE_TEAM = 'SELECT_YOUR_CHALLENGE_TEAM',
+  ACCEPT_INVITATION = 'ACCEPT_INVITATION',
   JOIN_REQUEST_FORM = 'JOIN_REQUEST_FORM',
   JOIN_REQUEST_SENT = 'JOIN_REQUEST_SENT',
   CREATE_NEW_TEAM = 'CREATE_NEW_TEAM',
   REGISTRATION_SUCCESSFUL = 'REGISTRATION_SUCCESSFUL',
 }
-type StepKey = keyof typeof StepsEnum
-type StepList = {
-  [key in StepKey]: Step
-}
 
-const steps: StepList = {
```

Comment:
Removed this entire `StepList` object, and removed usage of the `StepperDialog` component in this file.

My opinion is that the `StepList` was confusing because we would override some of these properties on-the-fly. The `StepperDialog` put a complicated interface on top of `DialogBase`, which exposed more complexity than it abstracted. I think it makes more sense to just use `DialogBase`.